src/main/kotlin/com/evalify/evalifybackend/semester/controller/SemesterController.kt

```
@@ -1,31 +1,110 @@
+        }
+    }
+
+    @DeleteMapping("/{semesterId}/courses/{courseId}")
+    fun deleteCourseFromSemester(
+        @PathVariable semesterId: UUID,
+        @PathVariable courseId: UUID
+    ): ResponseEntity<CourseResponse> {
+        return try {
+            val course = semesterService.deleteCourseFromSemester(semesterId, courseId)
+            ResponseEntity(course, HttpStatus.OK)
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)